lib/cocoapods/sandbox/path_list.rb

```diff
@@ -51,11 +51,22 @@ def read_file_system
           raise Informative, "Attempt to read non existent folder `#{root}`."
         end
         escaped_root = escape_path_for_glob(root)
-        absolute_paths = Pathname.glob(escaped_root + '**/*', File::FNM_DOTMATCH)
+
+        absolute_paths = Pathname.glob(escaped_root + '**/*', File::FNM_DOTMATCH).lazy
         dirs_and_files = absolute_paths.reject { |path| path.basename.to_s =~ /^\.\.?$/ }
-        relative_paths = dirs_and_files.map { |path| path.relative_path_from(root) }
-        sorted_paths = relative_paths.map(&:to_s).sort_by(&:upcase)
-        @dirs, @files = sorted_paths.partition { |path| File.directory?(root + path) }
+        dirs, files = dirs_and_files.partition { |path| File.directory?(path) }
+
+        root_length = root.cleanpath.to_s.length + File::SEPARATOR.length
+        relative_sorted = lambda do |paths|
```

Comment:
I want to make it clear that the lambda accepts full paths, but returns relative paths. `sort_relative_paths` sounds like it accepts relative paths. Maybe `sorted_relative_paths_from_full_paths` ?